apps/web/lib/team/[slug]/getServerSideProps.tsx

```diff
@@ -175,6 +176,14 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
     } as const;
   }
 
+  const disableOrgSubdomainURL = team?.parent?.organizationSettings?.disableOrgSubdomainURL;
+  if (isValidOrgDomain && disableOrgSubdomainURL) {
+    const redirect = getMainDomainOrgRedirect(context.req);
+    if (redirect) {
+      return redirect;
+    }
+  }
```

Comment:
We are going to have to identify the case where this route is for organization homepage(i.e. acme.cal.com) or a team page(e.g. acme.cal.com/marketing or cal.com/team1